### PR TITLE
[v2] Initial stab a separating into a Community Continuity Guidelines

### LINKED ISSUE #6: Proposal: Convert to "Community Continuity"
I feel like this name is closer to what we're getting at. This came out of a need for me + Ash to figure out how Moya should work properly, and to document this in a formalized way. 

The Moya Guidelines doc is a great example of how we think people should run projects. I want to make it easier to say you conform to this. There are three examples right now of conformity:
- Moya - it wrote, and owns the doc
- [Danger](http://github.com/danger/danger) - I added this to the README:
  
  > ## License, Contributor's Guidelines and Code of Conduct
  > 
  > This project is open source under the MIT license, which means you have full access to the source code and can modify it to fit your own needs.
  > 
  > This project subscribes to the [Moya Contributors Guidelines](https://github.com/Moya/contributors) which TLDR: means we give out push access easily and often.
  > 
  > Contributors subscribe to the [Contributor Code of Conduct](http://contributor-covenant.org/version/1/3/0/) based on the [Contributor Covenant](http://contributor-covenant.org) version 1.3.0.
- [Zewo](http://github.com/zewo/zewo) - who base their copy of their [CONTRIBUTING.md](https://github.com/Zewo/Zewo/blob/508703ca601be1f88d251d005059914f9de36985/CONTRIBUTING.md) from Moya's directly taking bits, and amending them for their own reasons.

Zewo's usage highlights a few flaws that I want to address:
- Background is specific to a project
- Names are obvious mine/Ash's
- There's specific bits around deployment
- Specifics around teams

I want to convert the Moya doc, into something that Danger/Zewo/Moya can all naturally conform to, and call that the "Community Continuity Doc." It'll never be as ubiquitous as a CoC, but I think it's a great pattern we should make easy to adopt. 

---

Separate to this, I think it might need to go into a new org eventually , and I think that this org should also make something like https://github.com/benbalter/squad_goals that automates the process of adding new people when their PRs have been merged. You could invite a specific GitHub account to your org, and it would handle the sending of the request, and maybe an email with a link to the CCD for that project. 

Dragging in @afeld who is a friend in the space, and might have ideas

### PR DESCRIPTION
**Pull request by orta**
Fixes #6 

---

OK, so, long time coming - thanks @paulofaria for our conversations recently. 

This splits out the README into the "moya/contributors README" and the `Community.md` file. The idea being that the README shows you how to get up and running, the `Community.md` is what you take into your own project and modify to your specifics. 

I'll be running through this for Danger, and the CocoaPods app ( both of which have different approaches to how this can be tackled ) before I can start to expand it further

@ashfurrow I want you to run through this for Moya/Moya - it's an oddity, but I think it makes sense. Plus having someone else run through the process will be helpful.

I think I'd advocate for renaming this repo to `community-continuity` too once this is merged and we've settled on what v2 is.

---

/cc @dbgrandi @ashfurrow @paulofaria @krausefx @segiddins @AliSoftware - would love feedback

What I'd like is focus on the new [Community.md](https://github.com/Moya/contributors/blob/v2/Community.md) which should feel reasonable in any project's repo.

### CODE REVIEW

File: Community.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+As the creators, and maintainers of this project. We want to ensure that the project lives and continues to grow. Not blocked by any singular person's computer time.
+
+One of the simplest ways of doing this, is by encouraging a larger set of shallow contributors.
+
+#### Ownership
+
+If you get a merged PR, regardless of content (typos, code, doc fixes) then you're eligible for push access to this organization. We check for this on PR merges and extend an invite via GitHub.
+
+Offhand, it's easy to imagine that this would make code quality suffer, but in reality it offers fresh perspectives to the codebase and encourages ownership from people who are depending on the project. If you are working with a codebase, then you probably have the skills to improve it, and offer valuable feedback.
```
Comment (alloy) [inline Community.md]: That last comma reads weird to me.

File: Community.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+This project is not continuously deployed, there is space for debate after review too. Offering the chance to revert, or make an amending pull request.
+
+It's convention to provide a call to action via `/cc @[org]/contributors` to invite discussion.
+
+#### How can we help you get comfortable contributing?
+
+It's normal for a first PR to be a potential fix for a problem, moving on from there to helping the project's direction can be difficult. We try to help contributors cross that barrier by offering good first step issues. These issues can be done without feeling like you're stepping on toes. Ideally, these are non-critical issues that are well defined. They will be purposely avoided by mature contributors to the project.
+
+We aim keep all project discussion inside GitHub issues. This is to make sure it is public. If you have questions about how to use the library, or how the project is running GitHub issues are the goto tool.
```
Comment (alloy) [inline Community.md]: > We aim **to**

Comment (alloy) [inline Community.md]: I think a comma is missing:

> …how to use the library, or how the project is running**,** GitHub issues are the goto tool.

And “how the project is running” sounds like a status update, do you mean “how the project is ran” ?

File: Community.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+#### Our expectations on you as a contributor
+
+To quote [@alloy](https://github.com/alloy) from [this issue](https://github.com/Moya/Moya/issues/135):
+
+> Don't ever feel bad for not contributing to open source.
+
+We want contributors to both provide ideas, keep the ship shipping and to take some of the load from others. It is non-obligatory; we're here to have fun. :trophy:
```
Comment (alloy) [inline Community.md]: ‘both’ implies 2 items only

Comment (alloy) [inline Community.md]: > we're here to have fun

That sounds more like a party. I would personally, if I were to add this to a repo of mine, change that to something like “we’re here to get things done in an enjoyable way”.

File: Community.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+If someone submits a pull request that's not perfect, and you are reviewing, it's better to think about the PR's motivation rather than the specific implementation. Having braces on the wrong line should not be a blocker. Though we do want to keep test coverage high, we will work with you to figure that out together.
+
+#### What about if you have problems that cannot be put into a public issue?
+
+Both [Ash Furrow](https://github.com/ashfurrow) and [Orta Therox](https://github.com/orta) have contactable emails on their GitHub profiles, and are happy to talk about any problems.
```
Comment (alloy) [inline Community.md]: Earlier in the doc you use an abstract example `/cc @[org]/contributors`, but here you use your real names. Is this doc meant to be an abstract template or the real Moya one?

Comment (orta) [inline Community.md]: No, I need to change this to be "here is an example version" - being that there needs to be a way that you can say you're available outside of public GH issues

File: README.md [hunk 5]
```diff
@@ -1,51 +1,48 @@
...
 After watching [The Social Coding Contract](http://blog.testdouble.com/posts/2014-12-02-the-social-coding-contract.html) we opted to find ways to make [the project](https://github.com/Moya/Moya/issues/135) more welcoming to external contributors.
 
-This document tries to clarify how we can include as many people as possible within the project.
-
-#### How to get push access?
-
-If you get a merged PR, regardless of content (typos, code, doc fixes) then you're eligible for push access to the Moya organization. We check for this on PR merges and extend an invite via GitHub.
-
-Offhand, it's easy to imagine that this would make code quality suffer, but in reality it offers fresh perspectives to the codebase and encourages ownership from people who are depending on the project.
+Because of the created the [Moya Community document](https://github.com/Moya/contributors/blob/0d5e80682b2377bdca72585eda9ce83467bee3c4/README.md) - over time this has been adopted in more projects.
```
Comment (AliSoftware) [inline README.md]: Should it really point to a blob and not on a tag or master?

Comment (orta) [inline README.md]: it was meant to show the original version, not the current, I can improve the wording around that

File: Code of Conduct.md [hunk 6]
```diff
@@ -1,48 +1,47 @@
...
 
 Examples of unacceptable behavior by participants include:
 
-* The use of sexualized language or imagery  
-* Personal attacks  
-* Trolling or insulting/derogatory comments  
-* Public or private harassment  
-* Publishing other's private information, such as physical or electronic  
-  addresses, without explicit permission  
+* The use of sexualized language or imagery
```
Comment (AliSoftware) [inline Code of Conduct.md]: - Mocking people for not knowing a feature of the project or of the language

Comment (orta) [inline Code of Conduct.md]: argh, I need to undo these changes. Definitely not doing a custom CoC. I strongly dislike the use of significant EOL whitespace in markdown.

Though I do agree with the idea 👍

**Comment by AliSoftware** [comment]
Side question, is there a way to automate the fact that a contributor is automatically given push access when the PR is merged?

It would also be cool to have a message to be automatically sent to thank them and explain that we granted them push access and what this means for them (e.g. no special expectation from them to give more time on the project — I've seen people _refusing_ the push access because their thought I expected them to be the next co-maintainer or something) + a link to the Community.md. That would probably be better achieved with the new "Saved Reply" feature of GitHub (rather than making a bot post that message for us) because a bot is less personal and posting that ourselves is more welcoming and would feel less robotic, but would still be nice to have a template for this kind of canned response, right?

File: Community.md (hunk 3 above)
Comment (AliSoftware) [inline Community.md]: Agreed I don't have a CONTRIBUTING.md file on my repo yet (that's one of the issues still open) so for now I just point to Moya's one until I find time to adapt to mine. But still, 2 of the last people I gave push access to, informing them in a comment in the PR, were quite "afraid" to receive that push access as in "no no no we won't have more time to spent on your project to help you maintain it, that was only a one-time PR", so I feel it might be interesting to add a paragraph here just to be extra-clear. Like:

> The fact that we will give you push access doesn't mean that we expect you to contribute more actively to the project or be more active. It's up to you to use that push access or don't use it at all, that's just a possibility we give you without any obligation whatsoever.
> 
> But the fact that you'll have push access would allow you to (1) avoid having to fork the project if you want to submit other PRs as you'll be able to create branches directly on the project (2) help triage issues, merge PRs, but it's up to you to use those superpowers or not :wink:

Thoughts?

Comment (AliSoftware) [inline Community.md]: _(or this paragraph can instead go in the saved response template I was talking about in my other comment, or both)_

File: README.md (hunk 5 above)
Comment (AliSoftware) [inline README.md]: Ok. The sentence was missing some words anyway 😆  ("Because of the created the…")

File: Community.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Community Continuity Guidelines v2.0.0
+
+As the creators, and maintainers of this project. We want to ensure that the project lives and continues to grow. Not blocked by any singular person's computer time.
```
Comment (dbgrandi) [inline Community.md]: `As the creators and maintainers of this project, we want to ...`

You could also change the second `.` to a `,` if you are okay with a 28 word sentence.

File: Community.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+One of the simplest ways of doing this, is by encouraging a larger set of shallow contributors.
```
Comment (dbgrandi) [inline Community.md]: `,` is not needed

File: Community.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+#### Ownership
+
+If you get a merged PR, regardless of content (typos, code, doc fixes) then you're eligible for push access to this organization. We check for this on PR merges and extend an invite via GitHub.
```
Comment (dbgrandi) [inline Community.md]: Is it limiting to mention GitHub specifically?

Comment (dbgrandi) [inline Community.md]: Missing a `,` after the `)` to close the `regardless of context` sub clause.

File: Community.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+Everyone comes in with their own perspective on what a project could/should look like, and encouraging discussion can help expose good ideas sooner.
+
+#### Why do we give out push access?
+
+It can be overwhelming to suddenly be offered the chance to wipe the source code for a project. Don't worry, we don't let you push to master.
+
+All code is peer-reviewed, and we have the convention that someone other than the submitter should merge the pull request.
+
+As an organization contributor you can merge other people's pull requests, or other contributors can merge yours. You won't be assigned a pull request, but you're welcome to just jump in and take a code review on things that interest you.
```
Comment (dbgrandi) [inline Community.md]: Missing comma  `As an organization contributor, you can...`

File: Community.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+If it feels right, merge.
+
+This project is not continuously deployed, there is space for debate after review too. Offering the chance to revert, or make an amending pull request.
```
Comment (dbgrandi) [inline Community.md]: This is a great thing to explicitly point out. Love it.

File: Community.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+It's convention to provide a call to action via `/cc @[org]/contributors` to invite discussion.
+
+#### How can we help you get comfortable contributing?
+
+It's normal for a first PR to be a potential fix for a problem, moving on from there to helping the project's direction can be difficult. We try to help contributors cross that barrier by offering good first step issues. These issues can be done without feeling like you're stepping on toes. Ideally, these are non-critical issues that are well defined. They will be purposely avoided by mature contributors to the project.
```
Comment (dbgrandi) [inline Community.md]: Add an `and` after the first comma to connect the thoughts more fluidly.

File: Code of Conduct.md (hunk 6 above)
Comment (ashfurrow) [inline Code of Conduct.md]: Might be worth updating to 1.4.0: http://contributor-covenant.org/version/1/4/code_of_conduct.md

File: Community.md (hunk 1 above)
Comment (ashfurrow) [inline Community.md]: I like "offhand" to describe this reaction; it's a common refrain we hear and using this phrasing frames it in a way that suggests its a reasonable feeling, even if the concern is unwarranted. I think it will help convince those on-the-fence to try this method of doing things.

File: Community.md (hunk 10 above)
Comment (ashfurrow) [inline Community.md]: > requests, or other

I think `requests, _and_ other` might make sense? Similarly, what about:

> You won't be assigned a pull request; instead, everyone is welcome...

File: Community.md (hunk 12 above)
Comment (ashfurrow) [inline Community.md]: Nice clarifying paragraph.

**Comment by orta** [comment]
> Side question, is there a way to automate the fact that a contributor is automatically given push access when the PR is merged?

[Danger as a web service](https://github.com/danger/danger/issues/42) could pull this off, for now the best option is to have checks like we have [in Danger](https://github.com/danger/danger/blob/0a26f1fa4e8b91b2296a3c0550c91017aeb2285c/Dangerfile#L7-L15) herself.

Agreed, we should provide advice on both using the Danger org check and a template for saved replies.

File: README.md [hunk 13]
```diff
@@ -1,51 +1,48 @@
...
+### I want to apply these guidelines
 
-All code is peer-reviewed, and we have the convention that someone other than the submitter should merge the pull request.
+Great! So, this is not quite another file to copy & paste into your project. It's slightly more involved.
 
-As an org contributor you can merge other people's pull requests, or other contributors can merge yours. You won't be assigned a pull request, but you're welcome to just jump in and take a code review on things that interest you.
+It's not too many steps though, and the majority of them are common sense things you would do anyway.
 
-If it feels right, merge. Moya is not continuously deployed, there is space for debate after too. Offering the chance to revert, or make an amending pull request.
+- [ ] Copy the Changelog.md into your repo
```
Comment (ashfurrow) [inline README.md]: Nice use of Markdown formatting, was initially confused 😄

**Comment by orta** [comment]
Everyone, thanks, I've agreed with all feedback, and made changes accordingly.

@dbgrandi - for now, I'm ok with it being GitHub specific (much as it pains me) there's a lot of GH specific terminology in here ( issues/ PR/ orgs) that could be cleaned up in another version at some point. 

For now it might make it more abstract and harder to grok overall.

@AliSoftware would linking to this section help now? https://github.com/Moya/contributors/blob/v2/Community.md#our-expectations-on-you-as-a-contributor

**Comment by ashfurrow** [comment]
The Contributor Covenant guidelines have an "attribution" note at the bottom linking back to them, making it easier for others to follow suit. Do you think having something similar pointing to this repo in `Community.md` makes sense?

File: Community.md (hunk 3 above)
Comment (dbgrandi) [inline Community.md]: @AliSoftware That's a great point. Invitation vs obligation can be a  subtle and scary difference in a lot of cases.

**Comment by orta** [comment]
> Do you think having something similar pointing to this repo in Community.md makes sense?

Yes

**Comment by AliSoftware** [comment]
> @AliSoftware would linking to this section help now? > https://github.com/Moya/contributors/blob/v2/Community.md#our-expectations-on-you-as-a-contributor

Indeed :+1:

File: README.md (hunk 13 above)
Comment (AliSoftware) [inline README.md]: There is no `CHANGELOG.md` in https://github.com/Moya/contributors/tree/v2 … so which file am I supposed to copy? 😆 

Didn't you mean `Community.md`? (and/or `README.md` & `Code of Conduct.md`?)

**Comment by AliSoftware** [comment]
Any reason why the file is not called `Contributing.md`?

I mean, I like that it's spelled lowercase instead of shouting-at-you-case 👍  but still as the common standard in OSS platforms like GitHub is to call the file serving exactly for this purpose (of explaining how it works for contributors) `CONTRIBUTING.md`, even if we prefer lowercase. At least keeping the name would help discoverability of that file/documentation, right?

**Comment by orta** [comment]
I think contributing.md has a different scope, as it's aimed at people writing issues and minutae for getting things working

https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md
https://github.com/realm/jazzy/blob/master/CONTRIBUTING.md
https://github.com/rails/rails/blob/master/CONTRIBUTING.md#how-to-contribute-to-ruby-on-rails
https://github.com/sinatra/sinatra/blob/master/CONTRIBUTING.md
https://github.com/npm/npm/blob/master/CONTRIBUTING.md

whereas this is more about explaining how you're going to encourage people to stay

**Comment by orta** [comment]
Given this a third run through, think it's pretty good to go.

**Comment by ashfurrow** [comment]
Will run through this one last time, apply to Moya, and report back.

**Comment by ashfurrow** [comment]
Got some time this afternoon, gonna prioritize this.

File: README.md [hunk 14]
```diff
@@ -1,51 +1,52 @@
...
+### I want to apply these guidelines
 
-#### When should you use push access?
+Great! So, this is not quite another file to copy & paste into your project. It's slightly more involved.
 
-It can be overwhelming to suddenly be offered the chance to wipe the source code for a project. Don't worry, we don't let you push to master.
+It's not too many steps though, and the majority of them are common sense things you would do anyway.
 
-All code is peer-reviewed, and we have the convention that someone other than the submitter should merge the pull request.
+- [ ] Copy the Changelog.md into your repo
```
Comment (ashfurrow) [inline README.md]: `Changelog.md` should be `Community.md`?

File: Community.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+- Pick up the project if other maintainers move their focus elsewhere.
+
+It's up to you to use those superpowers or not though 😉
+
+If someone submits a pull request that's not perfect, and you are reviewing, it's better to think about the PR's motivation rather than the specific implementation. Having braces on the wrong line should not be a blocker. Though we do want to keep test coverage high, we will work with you to figure that out together.
+
+#### What about if you have problems that cannot be discussed in a public issue?
+
+[ This is an example version, please delete this and replace with your own version ]
+Both [Ash Furrow](https://github.com/ashfurrow) and [Orta Therox](https://github.com/orta) have contactable emails on their GitHub profiles, and are happy to talk about any problems.
```
Comment (ashfurrow) [inline Community.md]: I think replacing the names/emails with placeholders might make it more clear what needs to be replaced.

File: README.md [hunk 16]
```diff
@@ -1,51 +1,52 @@
...
 
-If it feels right, merge. Moya is not continuously deployed, there is space for debate after too. Offering the chance to revert, or make an amending pull request.
+- [ ] Lock the master branch, to ensure code review as the way to get code in. Even for admins. Everyone plays by the rules.
 
-It's convention to provide a call to action [via](https://github.com/Moya/Moya/pull/273) `/cc @Moya/contributors` to invite discussion.
+- [ ] Ensure there is a label for people to find easy issues with
+  - We strongly recommend you use a [label name from one of this set](https://github.com/Charlotteis/libraries.io/blob/6afea1a3354aef4672d9b3a9fc4cc308d60020c8/app/models/github_issue.rb#L8-L14) in order to get larger appeal.
 
-#### How can we help you get comfortable contributing?
+- [ ] Fill out the "What about if you have problems that cannot be put into a public issue?" section at the end.
```
Comment (ashfurrow) [inline README.md]: This seems repeated from line 21 in this file.

File: Community.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+#### Where can I get more info about this document?
+
+The original source of this document can is at https://github.com/moya/contributors
```
Comment (ashfurrow) [inline Community.md]: Could be a link, with punctuation:

> The original source of this document can is at [https://github.com/moya/contributors](https://github.com/moya/contributors).

**Comment by ashfurrow** [comment]
I feel like maybe the adoption steps should discuss how to reference the document from the readme and in conversations.

**Comment by orta** [comment]
Could get some ideas from https://github.com/jekyll/jekyll/pull/5011 too

File: Community.md (hunk 17 above)
Comment (AliSoftware) [inline Community.md]: "can is"?

File: Community.md [hunk 18]
```diff
@@ -45,8 +45,8 @@ If someone submits a pull request that's not perfect, and you are reviewing, it'
 #### What about if you have problems that cannot be discussed in a public issue?
 
 [ This is an example version, please delete this and replace with your own version ]
-Both [Ash Furrow](https://github.com/ashfurrow) and [Orta Therox](https://github.com/orta) have contactable emails on their GitHub profiles, and are happy to talk about any problems.
+Both [Jane Bloggs](https://github.com/jane_b) and [Taylor Webb](https://github.com/twebb) have contactable emails on their GitHub profiles, and are happy to talk about any problems.
```
Comment (ashfurrow) [inline Community.md:48]: Is this intentional?

Comment (orta) [inline Community.md:48]: Yeah, generic names, all links go no-where

**Comment by orta** [comment]
Alright, I'm calling this 👍 

💇
